iroh-dns-server/src/http.rs

```diff
@@ -35,13 +36,61 @@ use crate::{config::Config, metrics::Metrics};
 
 pub use self::tls::CertMode;
 
+/// Config for http rate limit
+#[derive(Debug, Serialize, Clone)]
```

Comment:
Since serialization is not custom made, I think the serialization-deserialization roundrip might not work